integration_test/src/main.rs

```diff
@@ -0,0 +1,272 @@
+                    &oracle_pubkey,
+                    &oracle_r_value.key.into(),
+                    &local_fund_sk,
+                    &funding_script_pubkey,
+                    fund_output_value,
+                    z.1,
+                )
+                .expect("Error creating adaptor sig"),
+            )
+        });
+        let mut remote_cets_sigs = dlc_txs.cets.iter().zip(&msgs).map(|z| {
+            (
+                z,
+                dlc::create_cet_adaptor_sig(
+                    &secp,
+                    &z.0,
+                    &oracle_pubkey,
+                    &oracle_r_value.key.into(),
+                    &remote_fund_sk,
+                    &funding_script_pubkey,
+                    fund_output_value,
+                    z.1,
+                )
+                .expect("Error creating adaptor sig"),
+            )
+        });
+
+        assert!(local_cets_sigs.all(|z| dlc::verify_cet_adaptor_sig(
+            &secp,
+            &(z.1).0,
+            &(z.1).1,
+            &(z.0).0,
+            &oracle_pubkey,
+            &oracle_r_value.key.into(),
+            &local_params.fund_pubkey,
+            &funding_script_pubkey,
+            fund_output_value,
+            &(z.0).1
+        )
+        .is_ok()));
+
+        assert!(remote_cets_sigs
+            .clone()
+            .all(|z| dlc::verify_cet_adaptor_sig(
+                &secp,
+                &(z.1).0,
+                &(z.1).1,
+                &(z.0).0,
+                &oracle_pubkey,
+                &oracle_r_value.key.into(),
+                &remote_params.fund_pubkey,
+                &funding_script_pubkey,
+                fund_output_value,
+                &(z.0).1
+            )
+            .is_ok()));
+        remote_cets_sigs.nth(0).unwrap().1
+    };
+
+    let oracle_sig = secp
+        .schnorr_sign_with_nonce(&msgs[0], &oracle_sk.key, &oracle_k_value.key.into())
+        .expect("Error creating oracle signature.");
+
+    assert!(dlc::sign_cet(
+        &secp,
+        &mut dlc_txs.cets[0],
+        &remote_sig.0,
+        &oracle_sig,
+        &local_fund_sk,
+        &funding_script_pubkey,
+        fund_output_value,
+        true,
+    )
+    .is_ok());
+
+    dlc::util::sign_p2wpkh_input(
+        &secp,
+        &local_input_sk,
+        &mut dlc_txs.fund,
+        0,
+        &Address::p2pkh(&local_input_sk.public_key(&secp), Network::Regtest).script_pubkey(),
+        SigHashType::All,
+        local_params.input_amount,
+    );
+
+    dlc::util::sign_p2wpkh_input(
+        &secp,
+        &remote_input_sk,
+        &mut dlc_txs.fund,
+        1,
+        &Address::p2pkh(&remote_input_sk.public_key(&secp), Network::Regtest).script_pubkey(),
+        SigHashType::All,
+        remote_params.input_amount,
+    );
+
+    assert!(local_rpc.send_raw_transaction(&dlc_txs.fund).is_ok());
+
+    generate_blocks(1);
+
+    assert!(local_rpc.send_raw_transaction(&dlc_txs.cets[0]).is_ok());
```

Comment:
might be worthwhile to test the timelock enforcement on this one by setting it to something specific and then generating to that height (making a call to this with `.expect` first and then an assert after)

Reply:
Yeah timelock was definitely not working thanks... Fixed and added a test for that.